### DISCUSSION TITLE
LND Hub feature

### DISCUSSION THREAD

**Discussion by yannick10800** [btcpayserver/btcpayserver#1047] [General]
Hello
To allow liquidity for different shops using the same BTCpay server and to prevent each shop to open channels
Could you add the feature of LNDhub ? (such as https://github.com/dangeross/guides/blob/master/raspibolt/README.md)

To allow the Node to be a LND hub?
This allows user to be hub for different people, which reduces the cost of opening channels for end users (but is more centralized)
Thanks

**Comment by maltokyo** [comment]
Hi @Kukks 
I was thinking to make a PR to implement LNDHub into the docker version, for fans of Bluewallet.io, but:
1. I see you already self-assigned this to yourself
2. I see [you are looking](https://github.com/btcpayserver/btcpayserver/issues/204) at [other alternatives](https://github.com/lnbits/lnbits)

I tried to research this, but couldn't find the answer.  Does LNBits replace LNDHub as a drop-in replacement?

Without finding the above info, and after a chat with @NicolasDorier I made a docker for LNDhub here, with the intention of making a PR for BTCPay server here: https://github.com/maltokyo/docker-lndhub

It works well for me so far, but will not take it further (in terms of making PR for BTCPay) until I know your direction/thoughts on this.

**Comment by maltokyo** [comment]
Also, the developer of LNDHub ( @Overtorment ) left this comment, but I do not have the skill yet to know what it means, so not sure how to make progress:

https://github.com/BlueWallet/LndHub/issues/64#issuecomment-620713054_

**Comment by Kukks** [comment]
> Hi @Kukks
> I was thinking to make a PR to implement LNDHub into the docker version, for fans of Bluewallet.io, but:
> 
>     1. I see you already self-assigned this to yourself
> 
>     2. I see [you are looking](https://github.com/btcpayserver/btcpayserver/issues/204) at [other alternatives](https://github.com/lnbits/lnbits)
> 
> 
> I tried to research this, but couldn't find the answer. Does LNBits replace LNDHub as a drop-in replacement?
> 
> Without finding the above info, and after a chat with @NicolasDorier I made a docker for LNDhub here, with the intention of making a PR for BTCPay server here: https://github.com/maltokyo/docker-lndhub
> 
> It works well for me so far, but will not take it further (in terms of making PR for BTCPay) until I know your direction/thoughts on this.

Hey there, go for it :)

LNBits is not compatible with lndhub but it supports more implementations so I would aim to have both available. 

I can look into the api stuff in BTCPay so that it could work in a store once the docker image is merged

**Comment by maltokyo** [comment]
OK, got it, I will have a go, thanks!

**Comment by maltokyo** [comment]
Can anyone help me with this?

https://github.com/BlueWallet/LndHub/issues/95

**Comment by maltokyo** [comment]
@Kukks @NicolasDorier I'm a bit stuck with this.  Id really like to get LNDHub done before bwt.  Have some time this weekend, but cannot work this one out.  Would you mind to take a short look?

**Comment by maltokyo** [comment]
Here is my lndhub `config.js` file  (recycling the user and pw of other services on the same machine that I know work fine)

Everything look ok?  I give up for now, almost spent whole day trying to debug.  Maybe @NicolasDorier may know what caused these DNS resolution issues?

```
let config = {
  bitcoind: {
    rpc: 'http://lnd:afixedpasswordbecauselndsuckswithcookiefile@btcpayserver_bitcoind:43782',
  },
  redis: {
    port: 6379,
    host: 'lndhub_redis_btc',
    family: 4,
    password: '',
    db: 0,
  },
  lnd: {
    url: 'http://btcpayserver_lnd_bitcoin:8080/v1',
  },
};

if (process.env.CONFIG) {
  console.log('using config from env');
  config = JSON.parse(process.env.CONFIG);
}

module.exports = config;
```

Above config produces this in the log:

```
# docker logs lndhub_app_btc
using config {"bitcoind":{"rpc":"http://lnd:afixedpasswordbecauselndsuckswithcookiefile@btcpayserver_bitcoind:43782"},"redis":{"port":6379,"host":"lndhub_redis_btc","family":4,"password":"","db":0},"lnd":{"url":"http://192.168.1.38:8080/v1"}}
updateLightning()
updated
(node:19) DeprecationWarning: grpc.load: Use the @grpc/proto-loader module with grpc.loadPackageDefinition instead
(Use `node --trace-deprecation ...` to show where the warning was created)
2020-08-08T16:28:23.683Z : info: [BOOTING UP] : "Listening on port 3000"
lnd failure: Error: 14 UNAVAILABLE: DNS resolution failed
    at Object.exports.createStatusError (/lndhub/node_modules/grpc/src/common.js:91:15)
    at Object.onReceiveStatus (/lndhub/node_modules/grpc/src/client_interceptors.js:1209:28)
    at InterceptingListener._callNext (/lndhub/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/lndhub/node_modules/grpc/src/client_interceptors.js:618:8)
    at callback (/lndhub/node_modules/grpc/src/client_interceptors.js:847:24) {
  code: 14,
  metadata: Metadata { _internal_repr: {}, flags: 0 },
  details: 'DNS resolution failed'
}
lnd failure: Error: 14 UNAVAILABLE: DNS resolution failed
    at Object.exports.createStatusError (/lndhub/node_modules/grpc/src/common.js:91:15)
    at Object.onReceiveStatus (/lndhub/node_modules/grpc/src/client_interceptors.js:1209:28)
    at InterceptingListener._callNext (/lndhub/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/lndhub/node_modules/grpc/src/client_interceptors.js:618:8)
    at callback (/lndhub/node_modules/grpc/src/client_interceptors.js:847:24) {
  code: 14,
  metadata: Metadata { _internal_repr: {}, flags: 0 },
  details: 'DNS resolution failed'
}
lnd failure
Error: 14 UNAVAILABLE: DNS resolution failed
    at Object.exports.createStatusError (/lndhub/node_modules/grpc/src/common.js:91:15)
    at Object.onReceiveStatus (/lndhub/node_modules/grpc/src/client_interceptors.js:1209:28)
    at InterceptingListener._callNext (/lndhub/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/lndhub/node_modules/grpc/src/client_interceptors.js:618:8)
    at callback (/lndhub/node_modules/grpc/src/client_interceptors.js:847:24) {
  code: 14,
  metadata: Metadata { _internal_repr: {}, flags: 0 },
  details: 'DNS resolution failed'
}
```

**Comment by Overtorment** [comment]
I'm trying to help in https://github.com/BlueWallet/LndHub/issues/95
let me know if you need other assistance

**Comment by dennisreimann** [comment]
LNDHub has recently been added to the Docker deployment, see btcpayserver/btcpayserver-docker#472
